lib/baza-rb.rb

```diff
@@ -706,30 +706,29 @@ def checked(ret, allowed = [200])
       @loog.debug(log)
       return ret
     end
-    @loog.debug("#{log}\n  #{(ret.headers || {}).map { |k, v| "#{k}: #{v}" }.join("\n  ")}")
     headers = ret.headers || {}
-    msg = [
-      "Invalid response code ##{ret.code} ",
-      "at #{mtd} #{url}",
-      headers['X-Zerocracy-Flash'] ? " (#{headers['X-Zerocracy-Flash'].inspect})" : ''
-    ].join
+    if headers.any?
+      headers_log = headers.map { |k, v| "  #{k}: #{v}" }.join("\n")
+      @loog.debug("#{log}\n#{headers_log}")
+      @loog.error("#{log}\n#{headers_log}")
+    else
+      @loog.debug("#{log}\n  (no headers returned)")
+      @loog.error("#{log}\n  (no headers returned)")
+    end
+    info = []
+    info << "X-Zerocracy-Failure=#{headers['X-Zerocracy-Failure'].inspect}" if headers['X-Zerocracy-Failure']
+    if headers['X-Zerocracy-FailureMark']
+      info << "X-Zerocracy-FailureMark=#{headers['X-Zerocracy-FailureMark'].inspect}"
```

Comment:
@paveltrty this will make log line look pretty ugly. let's try to do better formatting.